sdcm/remote.py

```diff
@@ -213,6 +220,9 @@ def run(self, cmd, timeout=None, ignore_status=False,  # pylint: disable=too-man
             connect_timeout=300, verbose=True,
             log_file=None, retry=1, watchers=None):
         self.connection.connect_timeout = connect_timeout
+        if not self.is_up(timeout=connect_timeout):
```

Comment:
This introduced a deadlock.

_ssh_ping() won't success in executing 'true' command, it always fail in check is_up().
is_up event is set when _ssh_ping() succeeds.

Reply:
I don't understand how it can reach a deadlock, there is a `ssh_ping_thread` that is started when a remoter object is instantiated

Reply:
ssh_ping_thread() -> ssh_ping() -> run('true') -> is_up()  -> check is_up event

checking is_up event always fails.

Reply:
Do you mean `true` command will never succeed?

Reply:
ohh got it :)

Reply:
fixed! Thanks @amoskong